### DISCUSSION TITLE
Error when installing qpsolvers[open_source_solvers] with Python 3.7

### DISCUSSION THREAD

**Discussion by huaxia1985** [qpsolvers/qpsolvers#312] [Q&A] [answered]
Hello,
I have Python 3.7 and maxOS 10.15.4.
I got this error when installing qpsolvers[open_source_solvers]:
ERROR: Some build dependencies for cvxpy>=1.1.11 from https://files.pythonhosted.org/packages/f8/52/ab54781a450b44a5c6de20288bb623f37b85e636464af04cad90f5140c97/cvxpy-1.3.4.tar.gz (from qpsolvers[open_source_solvers]) conflict with the backend dependencies: numpy==1.14.5 is incompatible with numpy >= 1.15.

Then, I tried to install solvers with existing binaries:
install qpsolvers[wheels_only]
It was installed successfully. But when I tried to import qpsolvers, it raised ImportError that no QP solver found.
Thanks,
Xia

**Comment by stephane-caron** [comment, marked as answer]
Thank you for reporting this. Given that Python 3.7 has reached end of life (as a matter of fact 3.8 has done so as well: [Status of Python versions](https://devguide.python.org/versions/)), qpsolvers will likely require Python ≥ 3.9 in the near future.

Since you could install with wheels only, you likely have an older version of the project. Can you check which version was installed?

```console
$ python
Python 3.7.12 | packaged by conda-forge | (default, Oct 26 2021, 06:08:21) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import qpsolvers
>>> qpsolvers.__version__
'4.4.0'
```

At any rate, you can now try to install each solver individually. On my machine, with Python 3.7 and Ubuntu 22.04.5, the following ones worked:

- `pip install clarabel`
- `pip install cvxopt`
- `pip install daqp`
- `pip install ecos`
- `pip install highspy`
- `pip install osqp`
- `pip install piqp`
- `pip install proxqp`
- `pip install qpalm`
- `pip install quadprog`
- `pip install scs`

Hoping this helps!

**Comment by huaxia1985** [comment]
Thank you!
I updated python and the issue is resolved.
Xia
